Squash the build.

cstamas opened the pull request:
After recent changes (and near future ones, like the introduction of restlet1x plugin),
there is really no more reason to keep complicated module layout.

Changes:
- modules removed: nexus-utils, nexus-api, nexus-configuration, nexus-proxy, nexus-app, nexus-plugin-manager
- module "cleaned": nexus-configuration-model, now holds only modello models. Remains to be reusable (for some CLI tools)
- module nexus-core introduced, that is actually The Core of Nexus

No source changes happened, just move around. POMs were updated, removed all
occurrences of removed modules, introduced new module.

cstamas committed:
Squash the build.

After recent changes (and near future ones, like the introduction of restlet1x plugin),
there is really no more reason to keep complicated module layout.

Changes:
* modules removed: nexus-utils, nexus-api, nexus-configuration, nexus-proxy, nexus-app, nexus-plugin-manager
* module "cleaned": nexus-configuration-model, holds only modello models. Remains to be reusable (for some CLI tools)
* module nexus-core introduced, that is actually The Core of Nexus

No source changes happened, just move around. POMs were updated, removed all
occurrences of removed modules, introduced new module.

cstamas committed:
Added package that was ignored by gitignore.

This package should really be renamed.

cstamas committed:
Plugin Manager removed.

Left in by mistake, but it was properly pulled out
from the build in prev commit. Was unused.

adreghiciu commented:
+1 I hope that now we also get rid of duplicated packages in different modules. can you ensure that?

cstamas committed:
Merge remote-tracking branch 'origin/master' into squash-modules

Conflicts:
	nexus/nexus-api/pom.xml
	nexus/nexus-configuration/src/test/java/org/sonatype/nexus/configuration/application/SimpleApplicationConfiguration.java
	nexus/nexus-core/pom.xml
	nexus/pom.xml

cstamas committed:
Updates from master

cstamas committed:
Removing stale module

jdillon committed:
Merge branch 'master' into squash-modules

Conflicts:
	nexus/nexus-plugin-api/pom.xml

cstamas committed:
POM cleanup, during merge multiple entries for same dep creeped in.

cstamas committed:
Quickn dirty fix

This UT will fail whenever the count found during scan changes.

cstamas committed:
Added missing bits.

Now with squash, we have no "edition". So, adding some "UT Core"
edition and related files (are actually OSS edition copies) to be
present in UTs, as those are needed for nexus to boot.

cstamas commented:
Closing this pull, will be redone again by me, once master settles.
